src/gc-heap-snapshot.cpp

```diff
@@ -206,13 +207,18 @@ size_t record_node_to_gc_snapshot(jl_value_t *a) JL_NOTSAFEPOINT
         self_size = sizeof(jl_svec_t) + sizeof(void*) * jl_svec_len(a);
     }
     else if (jl_is_module(a)) {
-        name = "Module";
+        name = jl_symbol_name_(((_jl_module_t*)a)->name);
         self_size = sizeof(jl_module_t);
     }
     else if (jl_is_task(a)) {
         name = "Task";
         self_size = sizeof(jl_task_t);
     }
+    else if (jl_is_datatype(a)) {
+        type_name = string("Type{") + string(jl_symbol_name_(((_jl_datatype_t*)a)->name->name)) + string("}");
+        name = StringRef(type_name);
+        self_size = sizeof(jl_task_t);
+    }
```

Comment:
What was the output for datatypes before this PR? Was it just `DataType`?
I think that this should be improved even further after https://github.com/JuliaLang/julia/pull/47503.

Thanks! :)

Reply:
Yes, before this PR it was DataType, with it we got what "kind" of DataType and with that one we should get the specific type.